src/include/function/cast/functions/numeric_limits.h

```diff
@@ -79,5 +82,60 @@ constexpr uint64_t NumericLimits<double>::digits() {
     return 250;
 }
 
+template<typename T>
+static constexpr std::array<T, NumericLimits<T>::digits()> pow10Sequence() {
+    std::array<T, NumericLimits<T>::digits()> retval;
+    retval[0] = 1;
+    for (auto i = 1u; i < NumericLimits<T>::digits(); i++) {
+        retval[i] = retval[i - 1] * 10;
+    }
+    return retval;
+}
+
+template<>
+constexpr std::array<common::int128_t, NumericLimits<common::int128_t>::digits()> pow10Sequence() {
```

Comment:
I would suggest moving this function to cpp.

Reply:
constexpr functions need to be defined in the header